packages/@aws-cdk/aws-ec2/test/security-group.test.ts

```diff
@@ -148,6 +148,102 @@ nodeunitShim({
     test.done();
   },
 
+  'addIngressRule rule will not inline an inline-able rule if disableInlineRules is true'(test: Test) {
+    // GIVEN
+    const stack = new Stack();
+    const vpc = new Vpc(stack, 'VPC');
+
+    // WHEN
+    const sg = new SecurityGroup(stack, 'SG1', { vpc, allowAllOutbound: false, disableInlineRules: true })
+    sg.addEgressRule(Peer.anyIpv4(), Port.tcp(86), 'An external Rule');
+    
+    expect(stack).to(haveResource('AWS::EC2::SecurityGroup', {
+      GroupDescription: "Default/SG1",
+      VpcId: stack.resolve(vpc.vpcId)
+    }));
+
+    expect(stack).to(haveResource('AWS::EC2::SecurityGroupEgress', {
+          GroupId: stack.resolve(sg.securityGroupId),
+          CidrIp: '0.0.0.0/0',
+          Description: 'An external Rule',
+          FromPort: 86,
+          IpProtocol: 'tcp',
+          ToPort: 86,
+        },  
+    ));
+    expect(stack).to(not(haveResourceLike('AWS::EC2::SecurityGroupIngress', {}, )));
+    test.done();
+  },
+
+  'addIngressRule will not inline an inline-able rule if disableInlineRules is true'(test: Test) {
+    // GIVEN
+    const stack = new Stack();
+    const vpc = new Vpc(stack, 'VPC');
+
+    // WHEN
+    const sg = new SecurityGroup(stack, 'SG1', { vpc, allowAllOutbound: false, disableInlineRules: true })
+    sg.addIngressRule(Peer.anyIpv4(), Port.tcp(86), 'An external Rule');
+    
+    expect(stack).to(haveResource('AWS::EC2::SecurityGroup', {
+      GroupDescription: "Default/SG1",
+      VpcId: stack.resolve(vpc.vpcId)
+    }));
+
+    expect(stack).to(haveResource('AWS::EC2::SecurityGroupIngress', {
+          GroupId: stack.resolve(sg.securityGroupId),
+          CidrIp: '0.0.0.0/0',
+          Description: 'An external Rule',
+          FromPort: 86,
+          IpProtocol: 'tcp',
+          ToPort: 86,
+        },  
+    ));
+    expect(stack).to(not(haveResourceLike('AWS::EC2::SecurityGroupEgress', {}, )));
+    test.done();
+  },
+
+  'new SecurityGroup will not inline an allOutbound rule if disableInlineRules is true'(test: Test) {
+    // GIVEN
+    const stack = new Stack();
+    const vpc = new Vpc(stack, 'VPC');
+
+    // WHEN
+    const sg = new SecurityGroup(stack, 'SG1', { vpc, allowAllOutbound: true, disableInlineRules: true })
+    
+    expect(stack).to(haveResource('AWS::EC2::SecurityGroup', {
+      GroupDescription: "Default/SG1",
+      VpcId: stack.resolve(vpc.vpcId),
+    }));
+
+    expect(stack).to(haveResource('AWS::EC2::SecurityGroupEgress', {
+          GroupId: stack.resolve(sg.securityGroupId),
+          CidrIp: '0.0.0.0/0',
+          Description: 'Allow all outbound traffic by default',
+          IpProtocol: '-1',
+        },  
+    ));
+    expect(stack).to(not(haveResourceLike('AWS::EC2::SecurityGroupIngress', {}, )));
+    test.done();
+  },
+
+  'new SecurityGroup rule will not add any rules if allowAllOutbound is false and disableInlineRules is enabled'(test: Test) {
```

Comment:
This test case is kind of important.  I didn't build any handling for a NoOutbound egress rule when this feature is disabled.

I am considering traversing the stack and looking for NoOutbound egress rule in `removeNoTrafficRule` when `disableInlineRules` is true.

Reply:
The easiest way to handle this probably to store a reference to the ID of the egress rule object in some way, and then later on call `this.node.tryRemoveChild()` on it when it needs to be removed.